File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
class Solution {
    public int[] solution(int n, int m) {
        
        int[] answer = {gcd(n, m), lcm(n, m)};
        return answer;
    }
    
    private int gcd(int n, int m) {
        int max = Math.max(n, m);
        int min = Math.min(n, m);
        int r;
        
        while(min != 0) {
            r = max % min;
            max =min;
            min = r;
        }
        
        return max;
    }
    
    private int lcm(int n, int m) {
        return (n * m) / gcd(n, m);
    }
}

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.py
import math
def solution(n, m):
    return [get_gcd(n, m), get_lcm(n, m)]

def get_gcd(n, m):
    while m > 0:
        n, m = m, n % m
    return n

def get_lcm(n, m):
    return (n * m) / get_gcd(n, m)